xarray/tests/test_units.py

```diff
@@ -0,0 +1,1515 @@
+import operator
+from distutils.version import LooseVersion
+
+import numpy as np
+import pytest
+
+import xarray as xr
+from xarray.core.npcompat import IS_NEP18_ACTIVE
+
+pint = pytest.importorskip("pint")
+DimensionalityError = pint.errors.DimensionalityError
+
+
+pytestmark = [
+    pytest.mark.skipif(
+        not IS_NEP18_ACTIVE, reason="NUMPY_EXPERIMENTAL_ARRAY_FUNCTION is not enabled"
+    ),
+    # pytest.mark.filterwarnings("ignore:::pint[.*]"),
+]
+
+
+def use_pint_version_or_xfail(*, version, reason):
+    return pytest.mark.xfail(LooseVersion(pint.__version__) < version, reason=reason)
+
+
+unit_registry = pint.UnitRegistry()
+require_pint_array_function = pytest.mark.skipif(
+    not hasattr(unit_registry.Quantity, "__array_function__"),
+    reason="pint does not implement __array_function__ yet",
+)
```

Comment:
Same as above - the minimum version of pint supported by our policy - which means, the latest available - does support 0.9. So this whole check should go.

Reply:
the problem here is that the current pint master does not have `__array_function__` yet, so I think this is necessary to avoid failing too much.

Reply:
so the question is whether the whole module should be skipped if pint does not implement `__array_function__`, or if each individual test should be skipped (I think this makes a difference?)

Reply:
@keewis I think it's inconsequential from the moment we decided that we don't care about versions of pint that are older than the very latest available... if somebody runs the tests with an obsolete, unsupported version he'll get test failures.

Reply:
yeah, but the most recent version produces these failures right now (even with pint master). So until pint has a released version with `__array_function__`, I'd like to keep this

Reply:
@crusaderky I changed this to skipping the whole module, which should be removed once pint has at least one released version. Does this address your concern?

Reply:
fine for me